src/panels/calendar/ha-full-calendar.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  property,
+  PropertyValues,
+  LitElement,
+  CSSResult,
+  html,
+  css,
+  unsafeCSS,
+  query,
+} from "lit-element";
+import { Calendar } from "@fullcalendar/core";
+import dayGridPlugin from "@fullcalendar/daygrid";
+// @ts-ignore
+import fullcalendarStyle from "@fullcalendar/core/main.css";
+// @ts-ignore
+import daygridStyle from "@fullcalendar/daygrid/main.css";
+import "@material/mwc-icon-button";
+import "@material/mwc-menu";
+import "@material/mwc-button";
+import "@material/mwc-list/mwc-list-item";
+import type { Menu } from "@material/mwc-menu";
+import type { ListItemBase } from "@material/mwc-list/mwc-list-item-base";
+import type { Button } from "@material/mwc-button";
+
+import "../../components/ha-icon";
+
+import type { CalendarViewChanged, CalendarEvent } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+
+declare global {
+  interface HASSDomEvents {
+    "view-changed": CalendarViewChanged;
+  }
+}
+
+const fullCalendarConfig = {
+  headerToolbar: false,
+  plugins: [dayGridPlugin],
+  initialView: "dayGridMonth",
+  dayMaxEventRows: true,
+  height: "calc(100vh - 158px)",
+};
+
+class HAFullCalendar extends LitElement {
+  @property() public events: CalendarEvent[] = [];
+
+  @property() private calendar?: Calendar;
+
+  @property({ type: Boolean, reflect: true, attribute: "narrow" })
+  public narrow = false;
+
+  @query(".view-selection mwc-button") private _viewMenuButton?: Button;
+
+  protected render() {
+    return html`
+      <div class="card-content">
+        ${this.calendar
+          ? html`
+              <div class="header">
+                <div class="navigation">
+                  <mwc-button outlined @click=${this._handleToday}
+                    >Today</mwc-button
+                  >
+                  <mwc-icon-button
+                    class="prev"
+                    label="Prev"
+                    @click=${this._handlePrev}
+                  >
+                    <ha-icon icon="hass:chevron-left"></ha-icon>
+                  </mwc-icon-button>
+                  <mwc-icon-button
+                    class="next"
+                    label="Next"
+                    @click=${this._handleNext}
+                  >
+                    <ha-icon icon="hass:chevron-right"></ha-icon>
+                  </mwc-icon-button>
+                </div>
+                <div class="title">
+                  ${this.calendar.view.title}
+                </div>
+                <div class="view-selection">
+                  <mwc-button outlined @click=${this._handleViewButtonClick}>
+                    <ha-icon icon="hass:chevron-down"></ha-icon>
+                  </mwc-button>
+                  <mwc-menu activatable @selected=${this._handleView}>
```

Comment:
Why not use `mwc-select`?

Reply:
```
                <mwc-select outlined label="view" @change=${this._handleView}>
                  <mwc-list-item value="dayGridDay">Day</mwc-list-item>
                  <mwc-list-item value="dayGridWeek">Week</mwc-list-item>
                  <mwc-list-item selected value="dayGridMonth"
                    >Month</mwc-list-item
                  >
                </mwc-select>
```
```
  private _handleView(ev) {
    this.calendar!.changeView(ev.target.value);
    this._fireViewChanged();
  }
```

Reply:
I don't like the way the select looks. Doesn't look like it belongs there

Reply:
But I didn't use outlined. So maybe